src/components/Post/EditV2/Collections/CollectionSelectDropdown.tsx

```diff
@@ -0,0 +1,130 @@
+import { Divider, Group, Select, Stack, Text, Alert, Anchor } from '@mantine/core';
+import { CollectionMode } from '@prisma/client';
+import { useEffect, useMemo, useState } from 'react';
+import { useCollectionsForPostCreation } from '~/components/Collections/collection.utils';
+import { usePostEditParams, usePostEditStore } from '~/components/Post/EditV2/PostEditProvider';
+import { useDebouncer } from '~/utils/debouncer';
+import { showErrorNotification } from '~/utils/notifications';
+import { trpc } from '~/utils/trpc';
+import { isDefined } from '~/utils/type-guards';
+
+export const CollectionSelectDropdown = () => {
+  const { collections: queryCollectionIds, collectionId } = usePostEditParams();
+  const { post, updateCollection } = usePostEditStore((state) => ({
+    post: state.post,
+    updateCollection: state.updateCollection,
+  }));
+  const [selectedCollectionId, setSelectedCollectionId] = useState<number | null>(
+    post?.collectionId ?? collectionId ?? null
+  );
+  const [tagId, setTagId] = useState<number | null>(null);
+
+  const collectionIds = useMemo(() => {
+    return [...((queryCollectionIds as number[]) ?? []), collectionId, post?.collectionId].filter(
+      isDefined
+    );
+  }, [queryCollectionIds, collectionId, post]);
+
+  const { collections = [] } = useCollectionsForPostCreation({ collectionIds });
+  const writeableCollections = useMemo(() => {
+    return collections.filter(
+      (collection) => collection.permissions?.write || collection.permissions?.writeReview
+    );
+  }, [collections]);
+
+  const isContestCollectionsOnly = writeableCollections.every(
+    (collection) => collection.mode === CollectionMode.Contest
+  );
+
+  const selectedCollection = collectionId
+    ? writeableCollections.find((c) => c.id === collectionId)
+    : null;
+
+  useEffect(() => {
+    updateCollection(selectedCollectionId ?? null, null);
+    setTagId(null);
+  }, [selectedCollectionId]); // eslint-disable-line
+
+  useEffect(() => {
+    updateCollection(selectedCollectionId ?? null, tagId ?? null);
+  }, [tagId]);
```

Comment:
Instead of using `useEffect` for updating the collections, we should probably make the calls whenever we call `setCollectionId` or `setTagId`